drv/medusa-seq-api/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! API crate for the Medusa Sequencer server.
+
+#![no_std]
+
+use derive_idol_err::IdolError;
+use drv_fpga_api::FpgaError;
+use hubpack::SerializedSize;
+use serde::{Deserialize, Serialize};
+use userlib::{sys_send, FromPrimitive};
+
+#[derive(
+    Copy, Clone, Debug, PartialEq, Deserialize, Serialize, SerializedSize,
+)]
+pub enum RailName {
+    V1P0Mgmt,
+    V1P2Mgmt,
+    V2P5Mgmt,
+    V1P0Phy,
+    V2P5Phy,
+    V12QsfpOut,
+}
+
+#[derive(
+    Copy, Clone, Debug, FromPrimitive, Eq, PartialEq, IdolError, counters::Count,
+)]
+pub enum MedusaError {
+    FpgaError = 1,
+    NoFrontIOBoard,
+    FrontIOBoardPowerFault,
+    PowerFault,
```

Comment:
maybe a bit nitpicky but it might be nice to have a comment explaining the difference between these?